### PR TITLE
sosp userspace exercise: further refinement

### PR DESCRIPTION
**Pull request by daniel-scs**

### CODE REVIEW

**Comment by daniel-scs** [comment]
Okay, this is ready but needs somebody else to try it out.

**Comment by brghena** [comment]
Great changes, Daniel. I'm going through applications today and I'm going to manually merge these changes with some other changes I've got.

File: userland/examples/solutions/sosp-ble-ess.c [hunk 1]
```diff
@@ -58,14 +41,24 @@ int main(void)
...
+  temperature_read_sync(&temp);
+  update->type = SENSOR_TEMPERATURE;
+  update->value = temp;
   ipc_notify_svc(_svc_num);
 
-  return 0;
+  unsigned humi;
+  humidity_read_sync(&humi);
+  update->type = SENSOR_HUMIDITY;
+  update->value = temp;
```
Comment (bbbert) [inline userland/examples/solutions/sosp-ble-ess.c]: This should be `update->value = humi;`?
Also, I'm not entirely sure how IPC works in Tock, but if the callback isn't pre-emptive then won't there only be three humidity updates? I think we might need an asynchronous update loop.

Comment (brghena) [reply userland/examples/solutions/sosp-ble-ess.c]: Good call on both of these issues. I'm fixing them in the course-sosp branch right now

**Comment by brghena** [comment]
Alright, I pulled this all out into the branch. Thanks Daniel!

**Pull request closed without merging**